activerecord/test/cases/adapters/abstract_mysql_adapter/transaction_test.rb

```diff
@@ -149,5 +149,32 @@ class Sample < ActiveRecord::Base
       end
       assert_kind_of ActiveRecord::QueryAborted, error
     end
+
+    test "reconnect preserves isolation level" do
+      pool = Sample.connection_pool
+      @connection = Sample.lease_connection
+      first_begin_failed = false
+
+      @connection.singleton_class.define_method(:perform_query) do |raw_connection, sql, *args, **kwargs|
+        # Simulates the first BEGIN attempt failing
+        if sql.include?("BEGIN") && !first_begin_failed
+          first_begin_failed = true
+          raise ActiveRecord::ConnectionFailed, "Simulated failure"
+        end
+        super(raw_connection, sql, *args, **kwargs)
+      end
+
+      Sample.transaction(isolation: :read_committed) do
+        @connection.materialize_transactions
+        isolation = @connection.select_value("SELECT ISOLATION_LEVEL FROM performance_schema.events_transactions_current where THREAD_ID=PS_CURRENT_THREAD_ID()")
```

Comment:
This is MySQL 8 specific and looks to be be failing on CI :(

SELECT command denied to user 'rails'@'172.16.0.5' for table 'events_transactions_current'

Normally, you'd use something as simple as `select @@transaction_isolation` to check current isolation, but that one only works for session-wise transaction isolation setting.